lib/tools.js

```diff
@@ -139,11 +137,45 @@ function asJson (obj, msg, num, time) {
           value = (stringifier || stringify)(value)
       }
       if (value === undefined) continue
-      data += ',"' + key + '":' + value
+      propStr += ',"' + key + '":' + value
+    }
+  }
+
+  let msgStr = ''
+  if (msg !== undefined) {
+    value = serializers[messageKey] ? serializers[messageKey](msg) : msg
+    const stringifier = stringifiers[messageKey] || wildcardStringifier
+
+    switch (typeof value) {
+      case 'function':
```

Comment:
Should I cover all these cases here?

Reply:
If this is going to be left in, yes. But I'm really confused as to why it is needed.

Reply:
Well the `msg` key needs to be treated separatly so it can be placed on the root level and not the nested key one.